File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-5.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l=0;
        int sum=0;
        int minLen = 0;
        bool exist = false;
        for(int r=0; r<nums.size(); r++){
            sum += nums[r];
            while(sum>=target){
                if(!exist){
                    minLen = nums.size();
                }
                exist = true;
                if(r-l+1 < minLen){
                    minLen = r-l+1;
                }
                sum-=nums[l];
                l++;
            }
        }
        return minLen;
    }
};

File: Data Structures & Algorithms/search-in-rotated-sorted-array-ii/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> bool:
        l = 0
        r = len(nums)-1
        res = nums[0]
        while l<=r:
            m = l + ((r-l)//2)
            if nums[m]==target:
                return True
            elif nums[m]==nums[l]:
                l+=1
            elif nums[m] >= nums[l]:
                if target>= nums[l] and target<nums[m]:
                    r=m-1
                else:
                    l=m+1
            else:
                if target>nums[m] and target<=nums[r]:
                    l = m+1
                else:
                    r=m-1
        return False